Kitodo/src/main/java/org/kitodo/production/model/bibliography/course/Course.java

```diff
@@ -174,49 +299,108 @@ public Course(Document xml) {
+            yearStart = new MonthDay(dateTime.getMonthOfYear(), dateTime.getDayOfMonth());
+        }
+        yearName = rootNode.getAttribute(ATTRIBUTE_YEAR_TERM);
         Element processesNode = XMLUtils.getFirstChildWithTagName(rootNode, ELEMENT_PROCESSES);
         int initialCapacity = 10;
+        Map<LocalDate, IndividualIssue> lastIssueForDate = new HashMap<>();
+        List<RecoveredMetadata> recoveredMetadata = new LinkedList<>();
         for (Node processNode = processesNode.getFirstChild(); processNode != null; processNode = processNode
                 .getNextSibling()) {
             if (!(processNode instanceof Element) || !processNode.getNodeName().equals(ELEMENT_PROCESS)) {
                 continue;
             }
-            List<IndividualIssue> process =  prepareIndividualIssues(processNode, initialCapacity);
-            processes.add(process);
-            initialCapacity = (int) Math.round(1.1 * process.size());
-        }
-        recalculateRegularityOfIssues();
-        processesAreVolatile = true;
-    }
-
-    private List<IndividualIssue> prepareIndividualIssues(Node processNode, int initialCapacity) {
-        List<IndividualIssue> process = new ArrayList<>(initialCapacity);
-        for (Node blockNode = processNode.getFirstChild(); blockNode != null; blockNode = blockNode
-                .getNextSibling()) {
-            if (!(blockNode instanceof Element) || !blockNode.getNodeName().equals(ELEMENT_BLOCK)) {
-                continue;
-            }
-            String variant = ((Element) blockNode).getAttribute(ATTRIBUTE_VARIANT);
-            for (Node issueNode = blockNode.getFirstChild(); issueNode != null; issueNode = issueNode
+            List<IndividualIssue> process = new ArrayList<>(initialCapacity);
+            for (Node blockNode = processNode.getFirstChild(); blockNode != null; blockNode = blockNode
                     .getNextSibling()) {
-                if (!(issueNode instanceof Element) || !issueNode.getNodeName().equals(ELEMENT_APPEARED)) {
+                if (!(blockNode instanceof Element) || !blockNode.getNodeName().equals(ELEMENT_BLOCK)) {
                     continue;
                 }
-
-                IndividualIssue individualIssue = prepareIndividualIssue((Element) issueNode, variant);
-                process.add(individualIssue);
+                String variant = ((Element) blockNode).getAttribute(ATTRIBUTE_VARIANT);
+                for (Node issueNode = blockNode.getFirstChild(); issueNode != null; issueNode = issueNode
+                        .getNextSibling()) {
+                    if (!(issueNode instanceof Element) || !issueNode.getNodeName().equals(ELEMENT_APPEARED)) {
+                        continue;
+                    }
+                    String issue = ((Element) issueNode).getAttribute(ATTRIBUTE_ISSUE_HEADING);
+                    String date = ((Element) issueNode).getAttribute(ATTRIBUTE_DATE);
+                    if (date == null) {
```

Comment:
Should comparing to `null` or not to `null` to be made by `Objects.isNull(...)` and similar method calls? This is only one of the first places where this "old comparison" is done and I did not mark every single appearance in this whole pull request.